test2code/src/main/kotlin/com/epam/drill/plugins/test2code/Plugin.kt

```diff
@@ -256,21 +228,23 @@ class Plugin(
  * @features Coverage data sending
  */
 fun Plugin.probeSender(
-    sessionId: String,
     sendChanged: Boolean = false,
 ): RealtimeHandler = { execData ->
     execData
-        .map(ExecDatum::toExecClassData)
-        .chunked(0xffff)
-        .map { chunk -> CoverDataPart(sessionId, chunk) }
-        .sumOf { message ->
-            logger.trace { "send to admin-part '$message'..." }
-            val encoded = ProtoBuf.encodeToByteArray(CoverMessage.serializer(), message)
-            val compressed = Zstd.compress(encoded)
-            send(Base64.getEncoder().encodeToString(compressed))
-            message.data.count()
-        }.takeIf { sendChanged && it > 0 }?.let {
-            sendMessage(SessionChanged(sessionId, it))
+        .groupBy { it.sessionId }
+        .forEach { (sessionId, data) ->
+            data.map(ExecDatum::toExecClassData)
+                .chunked(0xffff)
+                .map { chunk -> CoverDataPart(sessionId, chunk) }
+                .sumOf { message ->
+                    logger.debug { "Send compressed message $message" }
+                    val encoded = ProtoBuf.encodeToByteArray(CoverMessage.serializer(), message)
+                    val compressed = Zstd.compress(encoded)
+                    send(Base64.getEncoder().encodeToString(compressed))
+                    message.data.count()
+                }.takeIf { sendChanged && it > 0 }?.let {
```

Comment:
- `sendChanged` mechanism purpose
- I believe the condition `it > 0` will always return `true` (since we must not send empty `message.data`)

Reply:
TODO - get rid of session changed on agent side, when current `admin` version is depreacted